src/modules/disk_used.rs

```diff
@@ -0,0 +1,307 @@
+        let disk_meta = fs::metadata(disk.get_name())?;
+        if disk_meta.rdev() == dev_id {
+            return Ok(disk);
+        }
+    }
+
+    Err(Box::new(DiskNotFoundError::new(path.to_owned())))
+}
+
+/// Uses the device id to get the drive
+#[cfg(target_os = "linux")]
+fn get_drive_from_path<'a>(path: &PathBuf, disks: &'a [Disk]) -> Result<&'a Disk, Box<dyn Error>> {
+    use std::{fs, os::linux::fs::MetadataExt};
+
+    let meta = fs::metadata(path)?;
+    let dev_id = meta.st_dev();
+
+    for disk in disks {
+        let disk_meta = fs::metadata(disk.get_name())?;
+        if disk_meta.st_rdev() == dev_id {
+            return Ok(disk);
+        }
+    }
+
+    Err(Box::new(DiskNotFoundError::new(path.to_owned())))
+}
+
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("disk_used");
+    let config = DiskUsedConfig::try_load(module.config);
+
+    if config.disabled {
+        return None;
+    }
+
+    let percentage_char = match context.shell {
+        Shell::Zsh => "%%", // % is an escape in zsh, see PROMPT in `man zshmisc`
+        _ => "%",
+    };
+
+    let mut current_disk = None;
+    let mut current_storage = None;
+    let mut other_storage: Option<Vec<Segment>> = None;
+    let system = System::new_with_specifics(RefreshKind::new().with_disks().with_disks_list());
+    let all_disks = system.get_disks();
+
+    if let Some(threshold) = config.current_threshold {
+        match get_drive_from_path(&context.current_dir, all_disks) {
+            Ok(disk) => {
+                current_disk = Some(disk);
+                if should_display_disk(disk, threshold) {
+                    match format_disk_used(
+                        &disk,
+                        &config,
+                        config.show_current_name,
+                        false,
+                        percentage_char,
+                    ) {
+                        Ok(segments) => {
+                            if !segments.is_empty() {
+                                current_storage = Some(segments);
+                            }
+                        }
+                        Err(e) => {
+                            log::warn!("Couldn't format disk from current path: {}", e);
+                        }
+                    }
+                }
+            }
+            Err(e) => {
+                log::warn!("Couldn't get disk from current path: {}", e);
+            }
+        }
+    }
+
+    if let Some(threshold) = config.all_threshold {
+        let display_disks: Vec<_> = all_disks
+            .iter()
+            .filter(|disk| match current_disk {
+                Some(current) => disk.get_name() != current.get_name(),
+                None => true,
+            })
+            .filter(|disk| should_display_disk(disk, threshold))
+            .collect();
+        let mut all_segments = Vec::new();
+
+        for (i, disk) in display_disks.iter().enumerate() {
```

Comment:
You don't need to change this but `filter_map` would work well here and avoid `collect`ing